src/js/hooks/productSupplier/form/useProductSupplierAttributes.js

```diff
@@ -0,0 +1,137 @@
+import { useEffect } from 'react';
+
+import { useDispatch, useSelector } from 'react-redux';
+import { z } from 'zod';
+
+import { fetchAttributes } from 'actions';
+import SelectField from 'components/form-elements/v2/SelectField';
+import TextInput from 'components/form-elements/v2/TextInput';
+import entityTypeCode from 'consts/entityTypeCode';
+import useTranslate from 'hooks/useTranslate';
+
+const useProductSupplierAttributes = () => {
+  const dispatch = useDispatch();
+  const translate = useTranslate();
+  const { attributes } = useSelector((state) => ({
+    attributes: state.productSupplier.attributes,
+  }));
+
+  useEffect(() => {
+    dispatch(fetchAttributes({
+      params: {
+        entityType: entityTypeCode.PRODUCT_SUPPLIER,
+      },
+    }));
+  }, []);
+
+  // When we have at least one option, it has to be select type
+  // (allowOther doesn't matter in this case)
+  const isSelectType = (attribute) => attribute?.options.length;
+
+  // When we don't have options, and allowOther is set to true, it is text input
+  const isTextType = (attribute) => attribute?.allowOther && !attribute?.options.length;
+
+  const inputTypeSchema = (attribute, errorProps) => (attribute.required
+    ? z
+      .string(errorProps)
+      .max(255, `Max length of ${attribute?.name} is 255`)
+    : z
+      .string()
+      .max(255, `Max length of ${attribute?.name} is 255`)
+      .optional()
+  );
+
+  const selectTypeSchema = (attribute, errorProps) => (attribute?.required
+    ? z.object({
+      value: z.string(),
+      label: z.string(),
+    }, errorProps)
+      .required()
+    : z.object({
+      value: z.string(),
+      label: z.string(),
+    })
+      .nullish()
```

Comment:
I think `.optional()` should also be added, otherwise it might still be considered as required, but I don't remember the exact case.

Reply:
`optional`, `nullable`, and `nullish` are designated to cover a specific value, undefined or null. The `nullish` covers all of those cases. So the `nullish` includes `optional` within itself.
```
{
    optional?: string | undefined;
    nullable: string | null;
    nullish?: string | null | undefined;
}
```